lib/elixir/lib/application.ex

```diff
@@ -364,7 +367,7 @@ defmodule Application do
   Note the environment is not returned as it can be accessed via
   `fetch_env/2`. Returns `nil` if the application is not loaded.
   """
-  @spec spec(app) :: [{key, value}] | nil
+  @spec spec(app) :: [{application_key, value}] | nil
```

Comment:
👍 

Btw, the [documentation in Erlang](http://erlang.org/doc/apps/kernel/application.html#get_all_key-1) only specifies `[{atom, term}]`, but that function only returns the keys specified above ([in the source code](https://github.com/erlang/otp/blob/da13166940df546d3fb05a71948f1709d72dd354/lib/kernel/src/application_controller.erl#L394-L412)).

Reply:
I will submit PR to OTP